### DISCUSSION TITLE
allowing the script to run on machines without SecureBoot and TPM v2

### DISCUSSION THREAD

**Discussion by atlantsecurity** [HotCakeX/Harden-Windows-Security#195] [General]
Many people host lots of stuff on virtual machines and situations where secureboot is disabled on purpose or when TPMv2 is not available abound. 
Would it be possible to create one without these requirements?

**Comment by HotCakeX** [comment]
Hi,
TPM 2.0 and Secure boot are old technologies, even computers running Intel 7th gen CPUs (maybe even older) support them. If they are disabled on purpose on a system, then they should be turned back on. Realistically, if someone intentionally compromises their own system by turning off security features, then naturally they don't care about security, so they don't need to apply these security measures.

TPM 2.0 and Secure boot are industry standards, basic yet fundamental security features and without them lots of security measures described on the readme simply don't make sense or are rendered useless, that's why there can't be a script that implements these security measures without meeting the underlying requirements, especially since they are the same requirements as the target OS they are built for.

Hyper-V VMs do support Secure boot and vTPM, whether they are hosted on Azure or locally. Might find [this article](https://github.com/HotCakeX/Harden-Windows-Security/wiki/Clean-Source-principle,-Azure-and-Privileged-Access-Workstations) interesting to read which is about clean source principle.

Sorry if it's not the response you were looking for, but security requires both hardware and software to work together.
